esp-storage/src/hardware.rs

```diff
@@ -3,18 +3,26 @@ use esp_rom_sys::rom::spiflash::*;
 
 use crate::maybe_with_critical_section;
 
+#[inline(never)]
+#[cfg_attr(not(target_os = "macos"), unsafe(link_section = ".rwtext"))]
```

Comment:
What's the macos bit for? Do we build this on hosted systems?

Reply:
We do host side checks on a Mac runner but good point - now that the link_section is moved to `hardware.rs` it's not needed anymore 👍